test/UnitTest/Microsoft.AspNet.OData.Test.Shared/SelectExpandNestedDollarCountTest.cs

```diff
@@ -112,9 +112,31 @@ public async Task SelectExpand_WithAllDollarCount_Works()
             Assert.Contains("\"Categories@odata.count\":9", payload); // Categories
         }
 
-        private Task<HttpResponseMessage> GetResponse(string uri, string acceptHeader)
+        [Fact]
+        public async Task SelectExpand_WithOneLevelNestedDollarCount_HandleNullPropagationFalse_Works()
         {
-            var controllers = new[] { typeof(MsCustomersController), typeof(MetadataController) };
+            // Arrange
+            string uri = "/odata/MsOrders?$expand=Categories($count=true)";
+
+            // Act
+            HttpResponseMessage response = await GetResponse(uri, AcceptJson);
+
+            // Assert
+            Assert.NotNull(response);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            string payload = await response.Content.ReadAsStringAsync();
+
+            Assert.DoesNotContain("\"Orders@odata.count\":5,", payload); // Top (Orders)
+            Assert.Contains("\"Categories@odata.count\":9", payload); // Categories
+        }
+
+        private Task<HttpResponseMessage> GetResponse(string uri, string acceptHeader, bool handleNullPropagation = true)
```

Comment:
>bool handleNullPropagation = true [](start = 87, length = 33)

Please add variations to the test for handle null propagation equal to both true and false (i.e., turn this method into [Theory] with [InlineData] for handleNullPropagation = true and handleNullPropagation=false). Better yet, make all of the tests that assume handleNullPropagation=false into [Theory] that run with handleNullPropagation equal to both true and false.

Reply:
removed the redundant argument - HandleNullPropagationOption is set on controller action level anyway

do not quite understand the point: SelectExpand_WithOneLevelNestedDollarCount_HandleNullPropagationFalse_Works is already a variation on test SelectExpand_WithOneLevelNestedDollarCount_Works which tests the same thing (count field for a nested collection) but with the default HandleNullPropagationOption value of True, which depends on the namespace of query provider (class HandleNullPropagationOptionHelper, method GetDefaultHandleNullPropagationOption)

Reply:
Right. My preference would be to have unit tests that explicitly tests both the null propagation true case as well as the null propagation false case. I realize that the existing test on ln 38 doesn't specify null propagation, which picks up the default, which for the current query provider happens to be null propagation true, but that's not explicit in the test so a future change to that test, such as using a different query provider, could inadvertently invalidate the intention that we have coverage for both null propagation false and null propagation true. 

One way to address this would be to specifically set HandleNullPropagation to OData.Query.HandleNullPropagationOption.True in the MsCustomersController, so that we are explicit that we are testing a handle null propagation value of true, rather than relying on a query provider-specific default value.

---
In reply to: [300698007](https://github.com/OData/WebApi/pull/1693#discussion_r300698007) [](ancestors = 300698007)